dev/io.openliberty.pages.3.1.internal.factories/src/org/apache/jasper/runtime/PageContextImpl.java

```diff
@@ -816,12 +816,19 @@ public ELContext getELContext() {
 		}
 
         // For Pages 3.1
-				if(servlet instanceof com.ibm.ws.jsp.runtime.JspDirectiveInfo && ((com.ibm.ws.jsp.runtime.JspDirectiveInfo)servlet).isErrorOnELNotFound() ){
-						this.elContext.putContext(jakarta.servlet.jsp.el.NotFoundELResolver.class, true);
-				}
-        // if( ((com.ibm.ws.jsp.runtime.HttpJspBase) servlet).getPropertyInfo().get("errorOnELNotFound") == true ){
-        //     this.elContext.putContext(jakarta.servlet.jsp.el.NotFoundELResolver.class, true);
-        // }
+		if(servlet instanceof com.ibm.ws.jsp.runtime.JspDirectiveInfo && ((com.ibm.ws.jsp.runtime.JspDirectiveInfo)servlet).isErrorOnELNotFound() ){
+			this.elContext.putContext(jakarta.servlet.jsp.el.NotFoundELResolver.class, true);
+		}
+
+        if(servlet instanceof com.ibm.ws.jsp.runtime.JspImportInfo){
+            for(String _package : ((com.ibm.ws.jsp.runtime.JspImportInfo)servlet).getImportPackageList()){
+                this.elContext.getImportHandler().importPackage(_package);
+            }
+            for(String _class : ((com.ibm.ws.jsp.runtime.JspImportInfo)servlet).getImportClassList()){
+                this.elContext.getImportHandler().importClass(_class);
+            }
+            // not sure about importing static fields/methods? spec doesn't seem to mention it? Issue 22507
```

Comment:
Can you expand on this comment? What issue is 22507, is it a spec issue, an open liberty issue, or something else?

In addition, are we not currently importing static fields/methods or are you saying we're not doing it since the spec doesn't say anything about it?

Reply:
Spec seems to be slient. Only information is this: 
```
An import attribute describes the types that are available to the scripting and EL
environments. The value is as in an import declaration in the Java programming
language, a (comma separated) list of either a fully qualified Java programming
language type name denoting that type, or of a package name followed by the .*
string, denoting all the public types declared in that package. The import list shall
be imported by the translated JSP page implementation and is thus available to
the scripting and EL environments.
```

Reply:
I updated it to mention liberty issue #22507

Reply:
I don't think there is anything to be done here. I don't think developers will import static fields in a page, but rather classes / packages. 

But I will follow up at a later time.